nrf-softdevice/src/flash.rs

```diff
@@ -71,25 +71,32 @@ impl ErrorType for Flash {
     type Error = FlashError;
 }
 
+impl ReadNorFlash for Flash {
+    const READ_SIZE: usize = 1;
+
+    fn read(&mut self, address: u32, data: &mut [u8]) -> Result<(), Self::Error> {
+        data.copy_from_slice(unsafe {
```

Comment:
could you keep the comments? especially the TODO is somewhat important :sweat_smile:

Reply:
Oops, sorry. I'll add that back in.

I've also found that implementing both traits is a little problematic because the method names clash, so you have to make calls like `ReadNorFlash::read(&mut flash, ...)`. Would it be better to provide sync inherent methods instead of implementing the trait?

Reply:
The conflict is also there with async vs blocking traits spi/uart/i2c/whatever read/write, and we impl all the traits anyway on the same struct.

 It's only an error if both traits are in scope, which should be rare: a given source code file will use async or blocking, but very rarely both.